src/main/java/org/broadinstitute/consent/http/resources/UserResource.java

```diff
@@ -136,6 +137,24 @@ public Response getDatasetsFromUserDacs(@Auth AuthUser authUser) {
         }
     }
 
+    @GET
+    @Path("/me/dac/datasets/v2")
+    @Produces("application/json")
+    @RolesAllowed({CHAIRPERSON, MEMBER})
+    public Response getDatasetsFromUserDacsV2(@Auth AuthUser authUser) {
+        try {
+            User user = userService.findUserByEmail(authUser.getEmail());
+            List<Integer> dacIds = user.getRoles().stream()
+                .map(UserRole::getDacId)
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
+            List<Dataset> datasets = dacIds.isEmpty() ? List.of() : datasetService.findDatasetListByDacIds(dacIds);
```

Comment:
Would it make sense to throw a 404 if the list is empty?

Reply:
Great point, I think that does make sense.